From 540eb374d63bcdf0bdb03f67e045d87850e46dbc Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:23:27 -0400
Subject: [tools] perf/build: split out feature check 'stackprotector'

Message-id: <1408462094-14194-39-git-send-email-jolsa@redhat.com>
Patchwork-id: 88022
O-Subject: [PATCH RHEL7.1 BZ1131394 038/325] tools/perf/build: Split out feature check: 'stackprotector'
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 430be5ab0a50a9116c689b2b2ea7acd7a635aabe
Author: Ingo Molnar <mingo@kernel.org>
Date: Mon Oct 7 09:47:00 2013 +0200
---
 create mode 100644 tools/perf/config/feature-checks/test-stackprotector.c

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index 1a67371..9f94912 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -104,6 +104,7 @@ $(info done)
 FEATURE_TESTS =    \
  hello    \
  stackprotector-all  \
+ stackprotector   \
  libnuma
 
 $(foreach test,$(FEATURE_TESTS),$(call feature_check,$(test),$(test)))
@@ -112,7 +113,7 @@ ifeq ($(feature-stackprotector-all), 1)
   CFLAGS += -fstack-protector-all
 endif
 
-ifeq ($(call try-cc,$(SOURCE_HELLO),$(CFLAGS) -Werror -Wstack-protector,-Wstack-protector),y)
+ifeq ($(feature-stackprotector), 1)
   CFLAGS += -Wstack-protector
 endif
 
diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index 7538c14..46b7650 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -2,6 +2,7 @@
 FILES=     \
  test-hello   \
  test-stackprotector-all  \
+ test-stackprotector  \
  test-libnuma
 
 CC := $(CC) -MD
@@ -18,6 +19,9 @@ test-hello:
 test-stackprotector-all:
  $(BUILD) -Werror -fstack-protector-all
 
+test-stackprotector:
+ $(BUILD) -Werror -fstack-protector
+
 test-libnuma:
  $(BUILD) -lnuma
 
diff --git a/tools/perf/config/feature-checks/test-stackprotector.c b/tools/perf/config/feature-checks/test-stackprotector.c
new file mode 100644
index 0000000..c9f398d
--- /dev/null
+++ b/tools/perf/config/feature-checks/test-stackprotector.c
@@ -0,0 +1,6 @@
+#include <stdio.h>
+
+int main(void)
+{
+ return puts("hi");
+}
-- 
1.7.1